flang/test/Fir/invliad-types.fir

```diff
@@ -0,0 +1,151 @@
+// Test the FIR types parser diagnostics
+// RUN: fir-opt -split-input-file -verify-diagnostics %s
+
+// expected-error@+1 {{expected non-function type}}
+func private @box3() -> !fir.boxproc<>
+
+// -----
+
+// expected-error@+2 {{expected non-function type}}
+// expected-error@+1 {{expected affine map}}
+func private @box1() -> !fir.box<!fir.array<?xf32>, >
+
+// -----
+
+// expected-error@+1 {{expected non-function type}}
+func private @box1() -> !fir.box<>
+
+// -----
+
+// expected-error@+1 {{expected integer value}}
+func private @box2() -> !fir.boxchar<>
+
+// -----
+
+// expected-error@+1 {{expected integer value}}
+func private @it6() -> !fir.char<>
+
+// -----
+
+// expected-error@+1 {{expected integer value}}
```

Comment:
Interesting consequence here. Either a `?` or an integer is ok, but it looks like the message falls from the control-flow of the parser. (This is merely an observation.)

Reply:
Yeah the `?` does not trigger a diagnostic and it falls on the next one which is the integer.